src/flint/fq_nmod.jl

```diff
@@ -49,6 +49,44 @@ function coeff(x::fq_nmod, n::Int)
                 (Ref{fq_nmod}, Int), x, n)
 end
 
+function coeffs_raw(x::fq_nmod)
+   @GC.preserve x begin
+      len = degree(parent(x))
+      V = unsafe_wrap(Vector{UInt}, reinterpret(Ptr{UInt}, x.coeffs), x.length)
+      Vcopy = Vector{UInt}(undef, len)
+      for i = 1:x.length
+         Vcopy[i] = V[i]
+      end
+      for i = x.length + 1:len
+         Vcopy[i] = 0
+      end
+   end
+   return Vcopy
+end
+
+function setindex_raw!(x::fq_nmod, c::UInt, i::Int)
+   len = degree(parent(x))
+   V = unsafe_wrap(Vector{UInt}, reinterpret(Ptr{UInt}, x.coeffs), len)
+   if i + 1 > x.length
+      if !iszero(c)
+         for j = x.length + 1:i
+            V[j] = 0
+         end
+         V[i + 1] = c
+         x.length = i + 1
+      end
+   else
+      V[i + 1] = c
+      if i + 1 == x.length && iszero(c)
+         while i >= 0 && iszero(V[i + 1])
+            x.length -= 1
+            i -= 1
+         end
+      end
+   end
+   return x
+end
```

Comment:
What is this function supposed to do?

Reply:
Read the ticket. It is an internal function requested for Gap interfacing.

Reply:
Fair enough, but the ticket doesn't mention the restrictions under which this function is safe. This function can easily crash an burn.

Reply:
I changed it.